src/Viewer.js

```diff
@@ -335,44 +335,67 @@ Viewer.prototype.domElement = function() {
  *        constructor.
  * @return {Scene}
  */
-Viewer.prototype.createScene = function(opts) {
+Viewer.prototype.createScene = function(MLopts) {
   // TODO: set textureStore size to 0 for video scenes somehow?
 
-  opts = opts || {};
+  MLopts = MLopts || {};
 
   var stage = this._stage;
+  var layers = [];
+
+  // Multilayer options now passed as Array of opts, wrapping in Array in case for single layer argument for backward compatibility
+  if( Object.prototype.toString.call( MLopts ) !== '[object Array]' ) MLopts = Array(MLopts);
+
+  MLopts.forEach(function(opts) {
+    
+    var source = opts.source;
+    var geometry = opts.geometry;
+    var view = opts.view;
+    var textureStore = new TextureStore(geometry, source, stage, opts.textureStore);
+    
+    var layer = new Layer(stage, source, geometry, view, textureStore, opts.layerOpts);
+  
+    layers.push(layer);
+  
+    if (opts.pinFirstLevel) {
+      layer.pinFirstLevel();
+    }
+    // Added option to pin specified level, keeping pinFirstLevel for backwards compatibility
+    if (opts.pinLevel) {
+      layer.pinLevel(opts.pinLevel);
+    }
+  }); 
 
-  var source = opts.source;
-  var geometry = opts.geometry;
-  var view = opts.view;
-  var textureStore = new TextureStore(geometry, source, stage, opts.textureStore);
-
-  var layer = new Layer(stage, source, geometry, view, textureStore, opts.layerOpts);
-
-  if (opts.pinFirstLevel) {
-    layer.pinFirstLevel();
-  }
-
-  var scene = new Scene(this, layer);
+  var scene = new Scene(this, layers);
   this._scenes.push(scene);
 
   return scene;
 };
 
 
-Viewer.prototype._addLayer = function(layer) {
-  // Pin the first level to serve as a last-resort fallback.
-  layer.pinFirstLevel();
-  this._stage.addLayer(layer);
+Viewer.prototype._addLayers = function(layers) {
+  var ref = this;
+  layers.forEach(function(layer, index) {
+    if(index === 0) {
+      // Pin the first level to serve as a last-resort fallback.
+      layer.pinFirstLevel();
+    } 
+    ref._stage.addLayer(layer);
+  }); 
 };
 
 
-Viewer.prototype._removeLayer = function(layer) {
-  if (this._stage.hasLayer(layer)) {
-    layer.unpinFirstLevel();
-    this._stage.removeLayer(layer);
-  }
-  layer.textureStore().clearNotPinned();
+Viewer.prototype._removeLayers = function(layers) {
+  var ref = this;
+  layers.forEach(function(layer, index) {
+    if (ref._stage.hasLayer(layer)) {
```

Comment:
The stage not having the layer anymore would mean there is a bug somewhere else in the code, right? I would remove the check, so that it crashes loudly.